src/error.rs

```diff
@@ -9,7 +16,59 @@ pub enum HcHttpGatewayError {
     /// Handles configuration parsing errors
     #[error("Configuration error: {0}")]
     ConfigurationError(String),
+    /// Handle path deserialization errors
+    #[error("Path deserialization error: {0}")]
+    PathDeserializationError(#[from] axum::extract::rejection::PathRejection),
```

Comment:
I'd call this something like `PathResolutionError` or `PathParsingError`.

Reply:
Renamed to the latter